src/test/java/htsjdk/samtools/cram/ref/CRAMReferenceRegionTest.java

```diff
@@ -167,6 +168,39 @@ public void testGetReferenceBasesByRegionPositive(
         Assert.assertEquals(bases, Arrays.copyOfRange(fullContigBases, requestedOffset, requestedOffset + requestedLength));
     }
 
+    // simulate the state transitions that occur when writing a CRAM file
+    @Test
+    public void testSerialStateTransitions() {
```

Comment:
Add a comment here explaining and referencing the bug that prompted us to write this test.

Reply:
Done.